CHANGELOG.md

```diff
@@ -1,3 +1,27 @@
+## v1.4.0 (20/03/2020)
+
+**Common**
+
+* The deprecated property `disk_path` was removed. Use the `disk_path_mutable` instead.
+* The `DefaultCache` constructor is now explicit and takes `CacheSettings` by value.
+* The bug that the data was not validated during the reading from leveldb when the `olp::cache::OpenOptions::CheckCrc` property provided fixed.
+* Various improvements in the `olp::http::NetworkCurl`.
+* Add a new method `SetDefaultHeaders` in `olp::http::Network`. Users can set default HTTP headers for each request made by the network. User agents set with default headers and user agents passed with network requests concatenated into one header.
+* Compiler warnings about deprecated methods/classes reduced.
+
+**olp-cpp-sdk-authentication**
+
+* The deprecated `AuthenticationClient::SignInClient` method was removed.
+
+**olp-cpp-sdk-dataservice-read**
+
+* `VersionedLayerClient` now triggers the error when the request is passed with `FetchOption::CacheWithUpdate`. It makes no sense to update the data when it is available in the cache for a `VersionedLayerClient` since the version is locked.
```

Comment:
```suggestion
* `VersionedLayerClient` now triggers an error when the request is passed with `FetchOption::CacheWithUpdate`. It makes no sense to update data when it is available in a cache for `VersionedLayerClient` since the version is locked.
```